spec/API_specification/creation_functions.md

```diff
@@ -60,14 +64,18 @@ Returns an uninitialized array having a specified `shape`.
 
     -   output array data type. If `dtype` is `None`, the output array data type must be the default floating-point data type. Default: `None`.
 
+-   **device**: _Optional\[ &lt;device&gt; ]_
+
+    -   device to place the created array on, if given. Default: `None`.
+
 #### Returns
 
 -   **out**: _&lt;array&gt;_
 
     -   an array containing uninitialized data.
 
 (function-empty_like)=
-### empty_like(x, /, *, dtype=None)
+### empty_like(x, /, *, dtype=None, device=None)
```

Comment:
This reminds me of an interest inquiry we had a while ago: Should the `_like*()` functions honor the device where the input array `x` is on? (https://github.com/cupy/cupy/pull/3457) 

Now I look back, it seems also plausible if the output array is on the same device as `x`, but my rejection still holds: it is incompatible with any sane device management approaches (context manager, explicit function call (such as `use_device(N)`), etc). I suppose having the newly added argument `device` will encounter the same challenge. The most radical example is `x` on device 1, the argument`device` is set to 2, but the current device is 0. 

Thoughts?

Reply:
> but my rejection still holds: it is incompatible with any sane device management approaches

I agree with the argument you made on that issue.

> I suppose having the newly added argument `device` will encounter the same challenge. The most radical example is `x` on device 1, the argument `device` is set to 2, but the default/current device is 0.

I don't see the conflict here. If a library has multiple ways of controlling device placement, the most explicit method should have the highest priority. So:
1. If `device=` keyword is specified, that always takes precedence
2. If `device=None`, then use the setting from a context manager, if set.
3. If no context manager was used, then use the global default device/strategy

Your example seems very similar to the first example of https://pytorch.org/docs/stable/notes/cuda.html#cuda-semantics, which I think explains desired behaviour here.

Reply:
The `empty_like` description seems clear enough: the `_like` is about `shape` and `dtype` only.

Reply:
Thanks, @rgommers I fully agree with everything you said above, but I noticed in the "Semantics" section that you're adding in this PR, the wording is a bit different and (arguably) less clear. Should we incorporate your above reply there? And it'd be nice to add a variant of "the `_like` is about `shape` and `dtype` only" emphasis to all the docstrings where `device` is an optional argument.

Reply:
Good ideas, done.